mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotation.java

```diff
@@ -27,61 +27,47 @@
 import com.google.common.base.Preconditions;
 
 public class MailboxAnnotation {
-
     private static final CharMatcher NAME_ANNOTATION_PATTERN = CharMatcher.ASCII
-            .and(CharMatcher.inRange('\u0000', '\u0019').negate()).and(CharMatcher.isNot('*'))
-            .and(CharMatcher.isNot('%'));
-    
+        .and(CharMatcher.inRange('\u0000', '\u0019').negate()).and(CharMatcher.isNot('*'))
+        .and(CharMatcher.isNot('%'));
+
     public static final String SLASH_CHARACTER = "/";
-    
+
     public static final String TWO_SLASH_CHARACTER = "//";
-    
-    public static MailboxAnnotation nil(String key) {
+
+    public static MailboxAnnotation nil(MailboxAnnotationKey key) {
         return new MailboxAnnotation(key, Optional.<String> absent());
     }
 
-    public static MailboxAnnotation newInstance(String key, String value) {
+    public static MailboxAnnotation newInstance(MailboxAnnotationKey key, String value) {
         return new MailboxAnnotation(key, Optional.of(value));
     }
 
-    private final String key;
+    private final MailboxAnnotationKey key;
     private final Optional<String> value;
 
-    private MailboxAnnotation(String key, Optional<String> value) {
+    private MailboxAnnotation(MailboxAnnotationKey key, Optional<String> value) {
         Preconditions.checkNotNull(key);
         Preconditions.checkNotNull(value);
-        Preconditions.checkArgument(isValidKey(key), 
-                "Key must start with '/' and not end with '/' and does not contain charater with hex from '\u0000' to '\u00019' or {'*', '%', two consecutive '/'} ");
+        Preconditions.checkArgument(key.isValid(),
```

Comment:
I may don't understand your question but the annotation should be valid before insert/update into store so we have to valid them.
Is that your question?